### PR TITLE
nginx-wsgi-flask - docker compose

### PR DESCRIPTION
**Pull request by GrantBirki**
# nginx-wsgi-flask

## About

Adding a simple and secure docker-compose sample for using NGINX + WSGI + Flask.

This sample demonstrates using a reverse proxy that points to a WSGI handler with a Flask backend. All images are based on Alpine, use nonroot Docker users, are validated by healthchecks, and are simple/easy to use.

## Note

This is a new PR that I created to resolve merge conflicts. It is based off this [pull request](https://github.com/docker/awesome-compose/pull/101)

### CODE REVIEW

**Review by aiordache — APPROVED** [review]
Thank you.

@glours Anything to add before merging this?

**Comment by GrantBirki** [comment]
@aiordache It is all good to be merged, thanks! :)

**Comment by glours** [comment]
@aiordache I'll try to take a look asap 👌

File: nginx-wsgi-flask/README.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+...
+Building nginx-proxy
+...
+Creating nginx-wsgi-flask_flask-app_1 ... done
+Creating nginx-wsgi-flask_nginx-proxy_1 ... done
+```
+
+## Expected result
+
+Listing containers must show three containers running and the port mapping as below:
```
Comment (glours) [inline nginx-wsgi-flask/README.md:54]: Should be two instead of `three containers running`

File: nginx-wsgi-flask/nginx/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
+FROM nginx:1.19.3-alpine
```
Comment (glours) [inline nginx-wsgi-flask/nginx/Dockerfile:1]: Can we upgrade to `1.19.7-alpine` to fix some known vulnerabilites?
```Target file:       ./nginx/Dockerfile
Project name:      docker-image|nginx-wsgi-flask_nginx-proxy
Docker image:      nginx-wsgi-flask_nginx-proxy:latest
Platform:          linux/amd64
Base image:        nginx:1.19.3-alpine
Licenses:          enabled

Tested 45 dependencies for known issues, found 8 issues.

Base Image           Vulnerabilities  Severity
nginx:1.19.3-alpine  8                4 high, 3 medium, 1 low

Recommendations for base image upgrade:

Minor upgrades
Base Image           Vulnerabilities  Severity
nginx:1.19.7-alpine  1                0 high, 1 medium, 0 low

File: nginx-wsgi-flask/nginx/nginx.conf [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    client_max_body_size 8m;
+    large_client_header_buffers 2 1k;
+
+    # # SSL / TLS Settings - Suggested for Security
+    # ssl_protocols TLSv1.2 TLSv1.3;
+	# ssl_session_timeout 15m;
+	# ssl_ciphers ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256;
+	# ssl_prefer_server_ciphers on;
+	# ssl_session_tickets off;
+
```
Comment (glours) [inline nginx-wsgi-flask/nginx/nginx.conf:47]: Should we keep this commented code? Is it a working config snippet for SSL?

**Review by glours — CHANGES_REQUESTED** [review]
Thanks for your contribution! 
Can you make some minor changes to be able of merge your PR.

File: nginx-wsgi-flask/nginx/nginx.conf (hunk 3 above)
Comment (GrantBirki) [reply nginx-wsgi-flask/nginx/nginx.conf:47]: Yep! That is a suggested block for using security best practices using TLS1.2 and higher. It also has a better TLS cipher suite for added security as well.

**Comment by GrantBirki** [comment]
@glours and @aiordache. I have made the changes above and they are in the new pull request available here: https://github.com/docker/awesome-compose/pull/120

I am currently traveling and needed to make a new branch in order to send the PR. Let me know if it all looks good, thanks! :)

**Pull request closed without merging**